### PR TITLE
Updated to compiler version 0.7.5.0 + Stronger types

### PR DESCRIPTION
**Pull request by FrigoEU**
Hey,

Just updated everything to the latest compiler version, plus added some stronger types + the streams package. Let me know if there's anything you don't like!

Simon

### CODE REVIEW

**Comment by joneshf** [comment]
Thanks! 

The merge went awry since the indentation changed in one of the files. Can you summarize what changed to make for better types?

Also same question here: https://github.com/joneshf/purescript-events/pull/2 (you can just answer it once, you don't have to answer it multiple times :))

**Comment by FrigoEU** [comment]
Hey,

As in the others, you can choose between the 0.1.0 and 0.1.0-rc.1 version of purescript-functions :).

**Comment by FrigoEU** [comment]
Oops, closed this by accident... reopening.

**Comment by FrigoEU** [comment]
I added some stronger types to the handling of events a child process can emit: Before, a ChildProcess was an EventEmitter and there were just some different kinds of events exposed. I removed that and just added a couple specific ffi functions (onExit, onMessage, ...) that take more strongly typed callbacks. It works a bit easier I think.

Simon

**Comment by joneshf** [comment]
Sounds good to me.
